### PR TITLE
Update HREF regex example

### LINKED ISSUE #25118: Incorrect regex given for matching hrefs
The example regex is incorrect because it does not correctly match unquoted hrefs which immediately precede the opening anchor tag's closing bracket.

Reproduction:

Input: `<a href=foo.com>Hello world</a>`

Your regex matches:  `foo.com>Hello` (Group 1)

Suggested fix:

Amend the regex to the following: `string HRefPattern = @"href\s*=\s*(?:[""'](?<1>[^""']*)[""']|(?<1>[^>\s]+))";`

Note that I have substituted your `\S+` at the end of the regex with `[^>\s]+`

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 130f000e-0ec1-e248-f804-631d3bda2711
* Version Independent ID: 25a9e501-343e-2ef4-05bc-1abc92b3ea96
* Content: [Regular Expression Example: Scanning for HREFs](https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-example-scanning-for-hrefs)
* Content Source: [docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md](https://github.com/dotnet/docs/blob/main/docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Ported code to new snippets format
- Cleaned up code variable names
- Changed snippet ID to be descriptive
- Fixed old doc porting problems like double spaces and indentation
- Updated table for regex explaination
- Used new regex suggested by @arctangent

Fixes #25118

@luisquintanilla

### CODE REVIEW

File: docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md [hunk 1]
```diff
@@ -14,46 +14,48 @@ helpviewer_keywords:
...
+| `href`                 | Match the literal string "href". The match is case-insensitive. |
+| `\s*`                  | Match zero or more white-space characters.                      |
+| `=`                    | Match the equals sign.                                          |
+| `\s*`                  | Match zero or more white-space characters.                      |
+| `(?:`                  | Start a non-capturing group.                                    |
+| `["'](?<1>[^"']*)["']` | Match a quotation mark or apostrophe, followed by a capturing group that matches any character other than a quotation mark or apostrophe, followed by a quotation mark or apostrophe. The group named `1` is included in this pattern. |
+| `|`                    | Boolean OR that matches either the previous expression or the next expression. |
+| `(?<1>[^>\s]+)`        | A capturing group that uses a negated set to match any character other than a greater-than sign or a whitespace character. The group named `1` is included in this pattern.|
+| `)`                    | End the non-capturing group.                                    |
```
Comment (adegeo) [inline docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md:47]: @luisquintanilla this is the only section really changed. Everything else was just formatting updates like removing doublespace at the end of lines and removing bad indentation.

File: docs/standard/base-types/snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+public class Class1
+{
+    // <main>
+    public static void Main()
+    {
+        string inputString = "My favorite web sites include:</P>" +
+                             "<A HREF=\"https://docs.microsoft.com/en-us/dotnet/\">" +
```
Comment (luisquintanilla) [inline docs/standard/base-types/snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs:10]: To confirm. The reason html tags and attributes are capitalized is to show that the regex expression is case insensitive?

**Review by luisquintanilla — APPROVED** [review]
@adegeo looks good to me. Just had a question re: code samples.

File: docs/standard/base-types/snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs (hunk 2 above)
Comment (adegeo) [reply docs/standard/base-types/snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs:10]: No, probably just old style............................................................................

**Pull request merged**